cpp/src/command_classes/CentralScene.cpp

```diff
@@ -162,34 +181,48 @@ bool CentralScene::HandleMsg
 	if( CentralSceneCmd_Set == (CentralSceneCmd)_data[0] )
 	{
 		// Central Scene Set received so send notification
-		int32 when;
-		if( _data[2] == 0 )
-			when = 0;
-		else if( _data[2] <= 0x7F )
-			when = _data[2];
-		else if( _data[2] <= 0xFE )
-			when = 60 * _data[2];
-		else
-			when = 0;
-		Log::Write( LogLevel_Info, GetNodeId(), "Received Central Scene set from node %d: scene id=%d in %d seconds. Sending event notification.", GetNodeId(), _data[3], when);
-
-		if( ValueInt* value = static_cast<ValueInt*>( GetValue( _instance, _data[3] ) ) )
+		uint8 keyAttribute = _data[2];
+		uint8 sceneID = _data[3];
+		Log::Write( LogLevel_Info, GetNodeId(), "Received Central Scene set from node %d: scene id=%d with key Attribute %d. Sending event notification.", GetNodeId(), sceneID, keyAttribute);
+
+		int index = 1;
+		if      ( keyAttribute == 0) { index = CentralScene_SinglePress; }
+		else if ( keyAttribute == 1) { index = CentralScene_Released; }
+		else if ( keyAttribute == 2) { index = CentralScene_HeldDown; }
+		else if ( keyAttribute == 3) { index = CentralScene_DoublePress; }
+		else if ( keyAttribute == 4) { index = CentralScene_TriplePress; }
+		else if ( keyAttribute == 5) { index = CentralScene_Pressed4Times; }
+		else if ( keyAttribute == 6) { index = CentralScene_Pressed5Times;}
+
+		if( ValueInt* value = static_cast<ValueInt*>( GetValue( sceneID, index ) ) )
 		{
-			value->OnValueRefreshed( when );
+			value->OnValueRefreshed( 1 );
 			value->Release();
 		} else {
 			Log::Write( LogLevel_Warning, GetNodeId(), "No ValueID created for Scene %d", _data[3]);
 			return false;
 		}
 		return true;
-	} else if (CentralSceneCmd_Capability_Report == (CentralSceneCmd)_data[0]) {
+	}
+	else if (CentralSceneCmd_Capability_Report == (CentralSceneCmd)_data[0])
+	{
 		/* Create a Number of ValueID's based on the m_scenecount variable
 		 * We prefer what the Config File specifies rather than what is returned by
 		 * the Device...
 		 */
 		int scenecount = _data[1];
-		if (m_scenecount != 0)
+		if (scenecount != 0)
+		{
 			m_scenecount = scenecount;
```

Comment:
your always testing if scenecount is not 0, and then setting against m_scenecount. This would override any config setting for scenecount (if we have to specify due to a buggy device). 
Testing m_scenecount != 0 instead allows us to override the the number of supported of scenes in a config file, and ignore anything the device sends.